ui/apps/platform/src/Components/CompoundSearchFilter/utils/utils.ts

```diff
@@ -0,0 +1,33 @@
+import { CompoundSearchFilterConfig, SearchFilterAttribute, SearchFilterEntity } from '../types';
+
+export function getEntities(config: Partial<CompoundSearchFilterConfig>): SearchFilterEntity[] {
+    const entities = Object.keys(config) as SearchFilterEntity[];
+    return entities;
+}
+
+export function getDefaultEntity(config: Partial<CompoundSearchFilterConfig>): SearchFilterEntity {
+    const defaultEntity = Object.keys(config)[0] as SearchFilterEntity;
+    return defaultEntity;
+}
```

Comment:
Nits: the chances of this happening are fairly small, since we have exported constants for the config values, but there are a couple of ways where we could end up with a non-`SearchFilterEntity` response here.
```ts
const conf = {
    BOGUS: [],
    Image: imageSearchFilterConfig,
};

const entity = getDefaultEntity(conf);
// typechecks, but `entity` is `"BOGUS"`
```

```ts
const conf = {};

const entity = getDefaultEntity(conf);
// typechecks, but `entity` is `undefined`
```
I think the only way to make this function type-safe-ish would be to 1. use a type guard and 2. allow `undefined` in the return type:
```ts
function isSearchFilterEntity(key: string): key is SearchFilterEntity {
   ...
}

export function getDefaultEntity(
    config: Partial<CompoundSearchFilterConfig>
): SearchFilterEntity | undefined {
    const entities = Object.keys(config).filter(isSearchFilterEntity);
    return entities[0];
}
```

We'd then have to handle the `undefined` case in the caller, and we would only be as type-safe as our implementation of `isSearchFilterEntity`, but I think the extra verbosity would make errors down the line less likely.

Reply:
At first, I was thinking about how that would work if the types limit the config object to specific entity names. Then I realized the config object wasn't typed before passing it to the function 🤦🏼‍♂️ . I guess it is true that we can't guarantee that the developer will add the proper type to the config object before passing it to the function. So this is an edge case to consider.